src/vcpkg/commands.build.cpp

```diff
@@ -1796,29 +1771,29 @@ namespace vcpkg
-            {"VCPKG_CMAKE_SYSTEM_VERSION", VcpkgTripletVar::CMAKE_SYSTEM_VERSION},
-            {"VCPKG_PLATFORM_TOOLSET", VcpkgTripletVar::PLATFORM_TOOLSET},
-            {"VCPKG_PLATFORM_TOOLSET_VERSION", VcpkgTripletVar::PLATFORM_TOOLSET_VERSION},
-            {"VCPKG_VISUAL_STUDIO_PATH", VcpkgTripletVar::VISUAL_STUDIO_PATH},
-            {"VCPKG_CHAINLOAD_TOOLCHAIN_FILE", VcpkgTripletVar::CHAINLOAD_TOOLCHAIN_FILE},
-            {"VCPKG_BUILD_TYPE", VcpkgTripletVar::BUILD_TYPE},
-            {"VCPKG_ENV_PASSTHROUGH", VcpkgTripletVar::ENV_PASSTHROUGH},
-            {"VCPKG_ENV_PASSTHROUGH_UNTRACKED", VcpkgTripletVar::ENV_PASSTHROUGH_UNTRACKED},
-            {"VCPKG_PUBLIC_ABI_OVERRIDE", VcpkgTripletVar::PUBLIC_ABI_OVERRIDE},
-            // Note: this value must come after VCPKG_CHAINLOAD_TOOLCHAIN_FILE because its default depends upon it.
-            {"VCPKG_LOAD_VCVARS_ENV", VcpkgTripletVar::LOAD_VCVARS_ENV},
-            {"VCPKG_DISABLE_COMPILER_TRACKING", VcpkgTripletVar::DISABLE_COMPILER_TRACKING},
-            {"VCPKG_XBOX_CONSOLE_TARGET", VcpkgTripletVar::XBOX_CONSOLE_TARGET},
-            {"Z_VCPKG_GameDKLatest", VcpkgTripletVar::Z_VCPKG_GameDKLatest},
+        static constexpr std::pair<StringLiteral, VcpkgTripletVar> VCPKG_OPTIONS[] = {
+            {CMakeVariableTargetArchitecture, VcpkgTripletVar::TARGET_ARCHITECTURE},
+            {CMakeVariableCMakeSystemName, VcpkgTripletVar::CMAKE_SYSTEM_NAME},
+            {CMakeVariableCMakeSystemVersion, VcpkgTripletVar::CMAKE_SYSTEM_VERSION},
+            {CMakeVariablePlatformToolset, VcpkgTripletVar::PLATFORM_TOOLSET},
+            {CMakeVariablePlatformToolsetVersion, VcpkgTripletVar::PLATFORM_TOOLSET_VERSION},
+            {CMakeVariableVisualStudioPath, VcpkgTripletVar::VISUAL_STUDIO_PATH},
+            {CMakeVariableChainloadToolchainFile, VcpkgTripletVar::CHAINLOAD_TOOLCHAIN_FILE},
+            {CMakeVariableBuildType, VcpkgTripletVar::BUILD_TYPE},
+            {CMakeVariableEnvPassthrough, VcpkgTripletVar::ENV_PASSTHROUGH},
+            {CMakeVariableEnvPassthroughUntracked, VcpkgTripletVar::ENV_PASSTHROUGH_UNTRACKED},
+            {CMakeVariablePublicAbiOverride, VcpkgTripletVar::PUBLIC_ABI_OVERRIDE},
+            // Note: this value must come after CMakeVariableChainloadToolchainFile because its default depends upon it.
+            {CMakeVariableLoadVcvarsEnv, VcpkgTripletVar::LOAD_VCVARS_ENV},
+            {CMakeVariableDisableCompilerTracking, VcpkgTripletVar::DISABLE_COMPILER_TRACKING},
+            {CMakeVariableXBoxConsoleTarget, VcpkgTripletVar::XBOX_CONSOLE_TARGET},
+            {CMakeVariableZVcpkgGameDKLatest, VcpkgTripletVar::Z_VCPKG_GameDKLatest},
         };
 
         const std::string empty;
         for (auto&& kv : VCPKG_OPTIONS)
         {
-            const std::string& variable_value = Util::value_or_default(cmakevars, kv.first, empty);
+            const auto& variable_value = Util::value_or_default(cmakevars, kv.first.to_string(), empty);
```

Comment:
This is not possible because `cmakevars` is `const std::unordered_map<std::string, std::string>&`, and `unordered_map` is not transparent. (If it were `std::map<std::string, std::string, std::less<>>` we could)

Reply:
See also https://github.com/microsoft/vcpkg-tool/pull/1362